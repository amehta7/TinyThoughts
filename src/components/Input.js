import React from 'react'
import PropTypes from 'prop-types'

const Input = ({ label, type, onInput, value = '', hasError }) => {
  return (
    <div className={hasError ? 'input-box hasError' : 'input-box'}>
      <span className='label'>{label}</span>
      <input type={type} value={value} onChange={onInput} />
    </div>
  )
}

Input.propTypes = {
  type: PropTypes.string,
}

export default Input
