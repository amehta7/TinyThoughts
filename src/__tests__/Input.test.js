import React from "react";
import { shallow } from "enzyme";
import Input from "../components/Input";

describe("Input component", () => {
  it("Renders a controlled component", () => {
    const onInputFn = jest.fn();
    const wrapper = shallow(
      <Input
        label="Username"
        type="email"
        onInput={onInputFn}
        value="joe@mockery.com"
        hasError={false}
      />
    );

    expect(wrapper).toMatchSnapshot();

    wrapper.find("input").simulate("change", { target: { value: "Try this" } });
    expect(onInputFn).toHaveBeenCalledTimes(1);
  });

  it("Renders a red border when hasError is set to true", () => {
    const wrapper = shallow(
      <Input
        label="Username"
        type="email"
        onInput={() => {}}
        value="joe@mockery.com"
        hasError={true}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
