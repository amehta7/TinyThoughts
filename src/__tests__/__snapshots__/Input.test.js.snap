// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component Renders a controlled component 1`] = `
<div
  className="input-box"
>
  <span
    className="label"
  >
    Username
  </span>
  <input
    onChange={[MockFunction]}
    type="email"
    value="joe@mockery.com"
  />
</div>
`;

exports[`Input component Renders a red border when hasError is set to true 1`] = `
<div
  className="input-box hasError"
>
  <span
    className="label"
  >
    Username
  </span>
  <input
    onChange={[Function]}
    type="email"
    value="joe@mockery.com"
  />
</div>
`;
